### ISSUE TITLE
How to connect unix socket file

### ISSUE THREAD

**Issue by ghost** [Gallopsled/pwntools#2090]
How to connect AF_UNIX

**Comment by Arusekk** [comment]
The way to do this is using [`remote.fromsocket`](https://docs.pwntools.com/en/dev/tubes/sockets.html#pwnlib.tubes.remote.remote.fromsocket). There is currently no wrapper, as UNIX socket exploits are apparently rare in CTFs.

You can first create a socket using [standard Python API](https://docs.python.org/dev/library/socket.html#socket-families)
```py
from pwn import *

s = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
s.connect('/run/your/unix/socket.sock')

t = tubes.fromsocket(s)

# your exploit code goes here

t.interactive()
```

Hope this helps!

**Comment by JustWhateverIOnlyWantBetterCodeView** [comment]
Workaround doesn't work anymore.

```
    host, port = s.getpeername()
ValueError: Too many values to unpack (expected 2)
```

**Comment by MrQubo** [comment]
It's `remote.fromsocket(s)` nowadays.

But workaround doesn't work anymore:
```
    host, port = s.getpeername()
ValueError: Too many values to unpack (expected 2)
```

**Comment by peace-maker** [comment]
I think that is fixed in #2497 too by accident.

**Comment by MrQubo** [comment]
> I think that is fixed in #2497 too by accident.

For AF_UNIX `getpeername()` returns a string. So I guess assuming that the path is not a single letter it might work? xd
